profiling/src/config.rs

```diff
@@ -0,0 +1,142 @@
+        }
+
+        /* If the sapi didn't have an env var, try the libc.
+         * Safety: pointer comes from valid CStr.
+         */
+        let val = libc::getenv(name.as_ptr() as *const c_char);
+        if val.is_null() {
+            return None;
+        }
+
+        /* Safety: `val` has been checked for NULL, though I haven't checked that
+         * `libc::getenv` always return a string less than `isize::MAX`.
+         */
+        let val = CStr::from_ptr(val);
+        let maybe_str = val.to_str().ok();
+        // treat empty strings the same as no string
+        maybe_str.filter(|str| !str.is_empty()).map(String::from)
+    }
+
+    /// # Safety
+    /// This can only be called during rinit.
+    pub unsafe fn get() -> Self {
+        let env = Self::getenv(CStr::from_bytes_with_nul_unchecked(b"DD_ENV\0"));
+        let profiling_enabled = Self::getenv(CStr::from_bytes_with_nul_unchecked(
+            b"DD_PROFILING_ENABLED\0",
+        ));
+        let profiling_log_level = Self::getenv(CStr::from_bytes_with_nul_unchecked(
+            b"DD_PROFILING_LOG_LEVEL\0",
+        ));
+
+        // This is the older, undocumented name.
+        let profiling_experimental_cpu_enabled = Self::getenv(CStr::from_bytes_with_nul_unchecked(
+            b"DD_PROFILING_EXPERIMENTAL_CPU_ENABLED\0",
+        ));
+        let profiling_experimental_cpu_time_enabled = Self::getenv(
+            CStr::from_bytes_with_nul_unchecked(b"DD_PROFILING_EXPERIMENTAL_CPU_TIME_ENABLED\0"),
+        );
+        let agent_host = Self::getenv(CStr::from_bytes_with_nul_unchecked(b"DD_AGENT_HOST\0"));
+        let trace_agent_port = Self::getenv(CStr::from_bytes_with_nul_unchecked(
+            b"DD_TRACE_AGENT_PORT\0",
+        ));
+        let trace_agent_url =
+            Self::getenv(CStr::from_bytes_with_nul_unchecked(b"DD_TRACE_AGENT_URL\0"));
+        let service = Self::getenv(CStr::from_bytes_with_nul_unchecked(b"DD_SERVICE\0"));
+        let version = Self::getenv(CStr::from_bytes_with_nul_unchecked(b"DD_VERSION\0"));
+        Self {
+            agent_host,
+            env,
+            profiling_enabled,
+            profiling_experimental_cpu_enabled,
+            profiling_experimental_cpu_time_enabled,
+            profiling_log_level,
+            service,
+            trace_agent_port,
+            trace_agent_url,
+            version,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub enum AgentEndpoint {
+    Uri(Uri),
+    Socket(PathBuf),
+}
+
+impl Default for AgentEndpoint {
+    /// Returns a socket configuration if the default socket exists, otherwise
+    /// it returns http://localhost:8126/. This does not consult environment
+    /// variables.
+    fn default() -> Self {
+        let path = Path::new("/var/run/datadog/apm.socket");
+        if path.exists() {
+            return AgentEndpoint::Socket(path.into());
+        }
+        AgentEndpoint::Uri(Uri::from_static("http://localhost:8126/"))
```

Comment:
```suggestion
        AgentEndpoint::Uri(Uri::from_static("http://localhost:8126"))
```
At least the tracer reports it without trailing slash, for consistency?